### PR TITLE
Automatically stop tours when server battles are stopped

### PR DESCRIPTION
**Pull request by ET54321**

### CODE REVIEW

**Comment by StrudelsPO** [comment]
er, tested right?

**Comment by Fuzzysqurl** [comment]
Idk what it does lol

**Comment by ET54321** [comment]
I partially tested it because it was late and I couldn't test much because I forgot to remove the "same ip" restriction. However, testing it now, this still doesn't pause the dq timers, so I'll work on this a bit more.

Basically, the idea was to automatically disable tours when /stopbattles is used for Battle Server updates by utilizing the global script.BattlesStopped flag.

**Comment by StrudelsPO** [comment]
you can probably just check for !tours.paused wherever removeinactive() is called

**Comment by ET54321** [comment]
Instead, I basically just paused the tours timer. This way, the time remaining for subs and inactivity stays the same when battles get turned back on.

**Comment by Fuzzysqurl** [comment]
To clarify: I knew what you were trying to do. I just didn't know the inner workings of tours to how that little code would have accomplished it.

**Comment by StrudelsPO** [comment]
is there a reason you need the paused property? looks like you can just check for !tours.working...

File: scripts/tours.js [hunk 1]
```diff
@@ -910,17 +911,28 @@ Used for things such as
...
+        }
+    }
     if (script.battlesStopped) {
         tours.paused = true;
+        return;
     } else {
         tours.paused = false;
+        for (var y2 in tours.tour) {
+            if (tours.tour[y2].time > systime) {
+                pausedTime = tours.tour[y2].time - systime;
```
Comment (StrudelsPO) [inline scripts/tours.js]: you're looping through all the tours, pausedTime is going to be continually overwritten :x

**Comment by ET54321** [comment]
It is there to stop /stoptours from breaking.  If I just used !tours.working, then if a tour owner needed to disable tours for some reason, it wouldn't stay disabled if battles weren't disabled.

**Comment by ET54321** [comment]
The systime is just code that I moved that was already there, but yea that is probably just redundant. Yea, pausedTime needs to be fixed.

**Comment by StrudelsPO** [comment]
just tour[x].time + 1 per tick while tours.paused should achieve what you need, that way the time limit moves forward with sys.time()

**Comment by ET54321** [comment]
How to do that so that it is once per tick?

**Comment by StrudelsPO** [comment]
um just something like
if (tours.paused) {
    for (var t in tours.tour) {
        if (tours.tour[t].time > sys.time()) {
            tours.tour[t].time += 1;
        }
    }
}
sys.time() increments 1 every second so with this your tour time also increments at the same pace during a pause. once you unpause there will be no net difference

**Comment by ET54321** [comment]
but isn't tourstep() going to be called far more frequently than once per second? so you'll end up incrementing tours time far more frequently than you need to

**Comment by StrudelsPO** [comment]
tourStep() is only called every 1 second

**Comment by StrudelsPO** [comment]
one more thing, maybe check if tours.tour[y1].time > sys.time before incrementing? functionally there shouldn't be a difference but it might affect tour history or records or whatever they have if you change the time of tours that are already past the limit

**Pull request merged**